modules/integration/tests-integration/tests-es/src/test/java/org/wso2/carbon/registry/es/notifications/CustomRXTStoreNotificationTestCase.java

```diff
@@ -0,0 +1,232 @@
+import org.wso2.greg.integration.common.utils.GREGIntegrationBaseTest;
+import org.wso2.greg.integration.common.utils.GenericRestClient;
+
+import javax.activation.DataHandler;
+import javax.ws.rs.core.MediaType;
+import javax.xml.xpath.XPathExpressionException;
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.testng.Assert.assertNotNull;
+
+/**
+ * This class test subscription & notification for custom rxt type at store.
+ */
+public class CustomRXTStoreNotificationTestCase extends GregESTestBaseTest {
+
+    private static final Log log = LogFactory.getLog(CustomRXTStoreNotificationTestCase.class);
+
+    private TestUserMode userMode;
+    String jSessionIdPublisher;
+    String jSessionIdStore;
+    String assetId;
+    String cookieHeaderPublisher;
+    String cookieHeaderStore;
+    GenericRestClient genericRestClient;
+    Map<String, String> queryParamMap;
+    Map<String, String> headerMap;
+    String publisherUrl;
+    String storeUrl;
+    String resourcePath;
+    private ResourceAdminServiceClient resourceAdminServiceClient;
+
+    @Factory(dataProvider = "userModeProvider")
+    public CustomRXTStoreNotificationTestCase(TestUserMode userMode) {
+        this.userMode = userMode;
+    }
+
+    @BeforeClass(alwaysRun = true)
+    public void init() throws Exception {
+        super.init(userMode);
+        String session = getSessionCookie();
+        genericRestClient = new GenericRestClient();
+        queryParamMap = new HashMap<String, String>();
+        headerMap = new HashMap<String, String>();
+        resourcePath =
+                FrameworkPathUtil.getSystemResourceLocation() + "artifacts" + File.separator + "GREG" + File.separator;
+        publisherUrl = automationContext.getContextUrls().getSecureServiceUrl().replace("services", "publisher/apis");
+        storeUrl = automationContext.getContextUrls().getSecureServiceUrl().replace("services", "store/apis");
+        resourceAdminServiceClient = new ResourceAdminServiceClient(backendURL, session);
+        addCustomRxt();
+        setTestEnvironment();
+    }
+
+    @Test(groups = { "wso2.greg",
+            "wso2.greg.es" }, description = "Adding subscription to custom asset on LC state change")
+    public void addSubscriptionToLcStateChange() throws JSONException, IOException {
+        JSONObject dataObject = new JSONObject();
+        dataObject.put("notificationType", "StoreLifeCycleStateChanged");
+        dataObject.put("notificationMethod", "work");
+
+        ClientResponse response = genericRestClient
+                .geneticRestRequestPost(storeUrl + "/subscription/applications/" + assetId, MediaType.APPLICATION_JSON,
+                        MediaType.APPLICATION_JSON, dataObject.toString(), queryParamMap, headerMap, cookieHeaderStore);
+
+        String payLoad = response.getEntity(String.class);
```

Comment:
Hi Amila,
Thanks for pointing this out. I have fixed this with PR #343 and apply the same fix for all the other places as well.
Thanks !